### PR TITLE
aws - ec2 - force stop override stop protection

### PR DESCRIPTION
**Pull request by gnokoheat**
- Problem
  - The force terminate function is applied, now (#7598)
  - But the force stop function is not applied yet. So, An error occurs when trying to stop instance has 'disableApiStop' attribute. There is no way with the current feature.
  - error log
```
botocore.exceptions.ClientError: An error occurred (OperationNotPermitted) when calling the StopInstances operation:
The instance 'i-03e56dee491732b73' may not be stopped. Modify its 'disableApiStop' instance attribute and try again.
[ERROR] 2022-11-08T08:28:58.128Z aac6a93f-1322-4f1b-84be-8bd715a96e29 error during policy execution

Traceback (most recent call last):
  File "/var/task/c7n/handler.py", line 165, in dispatch_event
    p.push(event, context)
  File "/var/task/c7n/policy.py", line 1288, in push
    return mode.run(event, lambda_ctx)
  File "/var/task/c7n/policy.py", line 583, in run
    return PullMode.run(self)
  File "/var/task/c7n/policy.py", line 364, in run
    results = a.process(resources)
  File "/var/task/c7n/resources/ec2.py", line 1490, in process
    self._run_instances_op(
  File "/var/task/c7n/resources/ec2.py", line 1498, in _run_instances_op
    return self.manager.retry(op, InstanceIds=instance_ids, **kwargs)
  File "/var/task/c7n/utils.py", line 446, in _retry
    return func(*args, **kw)
  File "/var/runtime/botocore/client.py", line 391, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/var/runtime/botocore/client.py", line 719, in _make_api_call
    raise error_class(parsed_response, operation_name)
```

- suggestion
  - stop instance function has also `force=true` options. And it tries to stop again after "modify instance attribute" when an error occurs due to "disableApiStop".

### CODE REVIEW

File: c7n/resources/ec2.py [hunk 1]
```diff
@@ -1476,31 +1482,66 @@ def process(self, instances):
...
+                if not self.data.get('force'):
+                    raise
+
+                self.log.info("Disabling stop and termination protection on instances")
+                self.disable_protection(
+                    client,
+                    op,
+                    [i for i in instances if i.get('InstanceLifecycle') != 'spot'],
+                )
+                return self.manager.retry(
```
Comment (kapilt) [inline c7n/resources/ec2.py:1541]: should this be here, we'll go back into the next iteration of the loop to try client_op

File: c7n/resources/ec2.py [hunk 2]
```diff
@@ -1476,31 +1482,66 @@ def process(self, instances):
...
+            if op == 'terminate':
+                modify_instance(i, 'disableApiTermination')
+
+        with self.executor_factory(max_workers=2) as w:
+            list(w.map(process_instance, instances, op))
+
+    def _run_instances_op(self, client, op, instances, **kwargs):
+        while instances:
+
+            client_op = client.stop_instances
```
Comment (kapilt) [inline c7n/resources/ec2.py:1519]: nit: outside of the loop on defs for client op

Comment (gnokoheat) [reply c7n/resources/ec2.py]: Yes, It's done.

File: c7n/resources/ec2.py (hunk 1 above)
Comment (gnokoheat) [reply c7n/resources/ec2.py]: Yes, I edited it. Is it okay ?

**Review by thisisshi — APPROVED** [review]
thanks, lgtm

**Pull request merged**